engine/table/src/main/java/io/deephaven/engine/table/impl/by/BaseChunkedCountOperator.java

```diff
@@ -104,7 +105,9 @@ private boolean removeChunk(
             int chunkSize) {
         final int count = doCount(chunkStart, chunkSize, values);
         if (count > 0) {
-            resultColumnSource.set(destination, plusLong(resultColumnSource.getUnsafe(destination), -count));
+            final long newCount = plusLong(resultColumnSource.getUnsafe(destination), -count);
+            Assert.geqZero(newCount, "newCount");
+            resultColumnSource.set(destination, newCount);
```

Comment:
Same logic should go on line 155 for modify chunk, because we can also incorrectly go negative there.